spec/models/spree/gateway/stripe_sca_spec.rb

```diff
@@ -3,80 +3,117 @@
+
+  let(:order) { create(:order_ready_for_payment) }
+
+  let(:year_valid) { Time.zone.now.year.next }
+
+  let(:credit_card) { create(:credit_card, gateway_payment_profile_id: pm_card.id) }
 
-  let(:order) { create(:order_with_totals_and_distribution) }
-  let(:credit_card) { create(:credit_card) }
   let(:payment) {
     create(
       :payment,
-      state: "checkout",
       order:,
       amount: order.total,
       payment_method: subject,
       source: credit_card,
-      response_code: "12345"
+      response_code: payment_intent.id
     )
   }
+
   let(:gateway_options) {
     { order_id: order.number }
   }
-  let(:payment_authorised) {
-    payment_intent(payment.amount, "requires_capture")
-  }
-  let(:capture_successful) {
-    payment_intent(payment.amount, "succeeded")
-  }
 
-  describe "#purchase" do
-    it "captures the payment" do
-      stub_request(:get, "https://api.stripe.com/v1/payment_intents/12345").
-        to_return(status: 200, body: payment_authorised)
-      stub_request(:post, "https://api.stripe.com/v1/payment_intents/12345/capture").
-        with(body: { "amount_to_capture" => order.total }).
-        to_return(status: 200, body: capture_successful)
+  describe "#purchase", :vcr, :stripe_version do
+    before { Stripe.api_key = secret }
+
+    let!(:pm_card) do
+      Stripe::PaymentMethod.create({
+                                     type: 'card',
+                                     card: {
+                                       number: '4242424242424242',
+                                       exp_month: 12,
+                                       exp_year: year_valid,
+                                       cvc: '314',
+                                     },
+                                   })
+    end
+    let!(:payment_intent) do
+      Stripe::PaymentIntent.create({
+                                     amount: 1000, # given in AUD cents
+                                     currency: 'aud', # AUD to match order currency
+                                     payment_method: pm_card,
+                                     payment_method_types: ['card'],
+                                     capture_method: 'manual',
+                                   })
+    end
+
+    # Stripe acepts amounts as positive integers representing how much to charge
+    # in the smallest currency unit
+    let(:capture_amount) { order.total.to_i * 10 }
+    let(:response) { subject.purchase(capture_amount, credit_card, gateway_options) }
+
+    before do
+      # confirms the payment
+      Stripe::PaymentIntent.confirm(payment_intent.id)
+    end
+
+    it "completes the purchase" do
+      payment
 
-      response = subject.purchase(order.total, credit_card, gateway_options)
+      expect {
+        response.to
+        change(
+          Stripe::PaymentIntent.retrieve(payment_intent.id).status
+        ).from("requires_capture").to("suceeded")
+      }
```

Comment:
I haven't seen this format before, I think it's meant to be:
```suggestion
      expect {
        response
      }.to change(
        Stripe::PaymentIntent.retrieve(payment_intent.id).status
      ).from("requires_capture").to("suceeded")
```

Reply:
Nice catch! And Interesting question...

Doing so throws:
```
     ArgumentError:
       `change` requires either an object and message (`change(obj, :msg)`) or a block (`change { }`). You passed an object but no message.
```

So I've tried calling :statue as the error message indicates:

```
      expect{response}.to change(
          Stripe::PaymentIntent.retrieve(payment_intent.id), :status).from("requires_capture").to("suceeded")
```

Doing so makes the spec fail:

```
expected `Stripe::PaymentIntent#status` to have changed from "requires_capture" to "suceeded", but did not change
```

This all makes sense to me now. In fact, this was a test I've included for my better understanding. I see now it was passing because we set no real expectation... `expect{}` just returns the object... 

So, why does the spec fail if this is done correctly?
Because response does not change the `payment_intent.status` (as I thought it would)... Even if it would, I don't think I could check it in this way, because afaik, VCR only records one tape per example. We would need to make sure we can compare the same payment intent, before and after we call `purchase()`.

`purchase()` does call `capture()` -> this is the reason why I was expecting this test to pass. But it is not the case: I've checked the Stripe dashboard for that payment intent and it is not the case.

It is only if I run `Stripe::PaymentIntent.capture("pi_3OMVLCKuuB1fWySn2HxuWR75")` that the status changes to `succeeded`.

I don't think this is a bug. This was my (incorrect) reading of what should happen when `purchase()` is called, over the gateway. Instead, the response is actually an ActiveMerchant object. 

I'll change this to keep the original assertion only.

Thanks so much @dacook !